### PR TITLE
Feature/298 support browsers without localstorage

### LINKED ISSUE #298: Support browsers without LocalStorage support
e.g. Mobile Safari 10 does not seem to support LocalStorage which provokes an exception

https://sentry.pegnu.cloud/pg/oc/issues/210/

```
SecurityError: The operation is insecure.
  at getItem (/opt/jenkins-agent/home/workspace/ra_outstanding-cobra_master-ANCNH4WN233VFFGAWBI23ICJLX2M4GHW3RZ7J6NQOUNYLRNCD56Q/frontend/src/app/app.component.ts:93:35)
  at getThemeFromLocalStorage (/opt/jenkins-agent/home/workspace/ra_outstanding-cobra_master-ANCNH4WN233VFFGAWBI23ICJLX2M4GHW3RZ7J6NQOUNYLRNCD56Q/frontend/src/app/app.component.ts:43:30)
  at ngOnInit (/opt/jenkins-agent/home/workspace/ra_outstanding-cobra_master-ANCNH4WN233VFFGAWBI23ICJLX2M4GHW3RZ7J6NQOUNYLRNCD56Q/frontend/node_modules/@angular/core/@angular/core.es5.js:10836)
  at view (/opt/jenkins-agent/home/workspace/ra_outstanding-cobra_master-ANCNH4WN233VFFGAWBI23ICJLX2M4GHW3RZ7J6NQOUNYLRNCD56Q/frontend/node_modules/@angular/core/@angular/core.es5.js:12332)
  at view (/opt/jenkins-agent/home/workspace/ra_outstanding-cobra_master-ANCNH4WN233VFFGAWBI23ICJLX2M4GHW3RZ7J6NQOUNYLRNCD56Q/frontend/node_modules/@angular/core/@angular/core.es5.js:12965)
...
(19 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by jmesserli**
This closes #298.

Summary of changes:
- ​Add basil.js library which wraps localStorage, cookie, sessionStorage and in-memory options
- Use it everywhere localStorage was used

I made sure to:
- [x] Run the code formatter (`CTRL + ALT + L`)

Please review @M4R1KU: @outcobra/developers.

### CODE REVIEW

File: frontend/src/app/layout/app-layout/app-layout.component.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import {AfterViewInit, Component, HostBinding, OnInit, ViewChild} from '@angular/core';
 import {DefaultAuthService} from '../../core/services/auth/auth.service';
-import {NavigationEnd, NavigationStart, Router} from '@angular/router';
```
Comment (needToRoll) [inline frontend/src/app/layout/app-layout/app-layout.component.ts]: Did you remove this on purose?

Comment (jmesserli) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: It is not used

Comment (needToRoll) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: Ok then, it just seemed a little off-topic

Comment (jmesserli) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: Have you heard of `CTRL + ALT + L`? 😂

Comment (M4R1KU) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: Most likeyl Ctrl+Alt+O

Comment (needToRoll) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: Yes, I have but the point of a review is to look at the code and look for everything that seems suspicious. This is the only way we can prevent bugs.

Comment (M4R1KU) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: Ctrl+Alt+L  does not remove imports, doesn't it?

Comment (jmesserli) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: @needToRoll I agree but checking imports is the job of CI not code review.
@M4R1KU Yes it does if you enable optimize imports as part of it

Comment (needToRoll) [reply frontend/src/app/layout/app-layout/app-layout.component.ts]: @M4R1KU @jmesserli you can configure the behaviour of the code formatter in your settings. You cand define wathever you want the imports to be optimized or not.

**Pull request merged**